be/src/exec/aggregator.cpp

```diff
@@ -804,37 +804,53 @@ Status Aggregator::_evaluate_const_columns(int i) {
 
 Status Aggregator::convert_to_chunk_no_groupby(ChunkPtr* chunk) {
     SCOPED_TIMER(_agg_stat->get_results_timer);
-    // TODO(kks): we should approve memory allocate here
-    auto use_intermediate = _use_intermediate_as_output();
-    Columns agg_result_column = _create_agg_result_columns(1, use_intermediate);
-    if (!use_intermediate) {
+    if (_agg_functions[0]->get_name() == "group_concat2") {
+        Columns agg_result_column = _create_agg_result_columns(_state->chunk_size(), false);
         TRY_CATCH_BAD_ALLOC(_finalize_to_chunk(_single_agg_state, agg_result_column));
+        RETURN_IF_ERROR(check_has_error());
+
+        TupleDescriptor* tuple_desc = _output_tuple_desc;
+        ChunkPtr result_chunk = std::make_shared<Chunk>();
+        for (size_t i = 0; i < agg_result_column.size(); i++) {
+            result_chunk->append_column(std::move(agg_result_column[i]), tuple_desc->slots()[i]->id());
+        }
+        auto result_rows = result_chunk->num_rows();
+        _num_rows_returned += result_rows;
+        _num_rows_processed += result_rows;
+        *chunk = std::move(result_chunk);
+        _is_ht_eos = (result_rows == 0);
```

Comment:
This question is related to the question in line#632 in group_concat.h.
I expected finalize_to_column() was called once with all aggregated data and this line was `_is__ht_eos = true;`. Why is it not the case?

Reply:
you can still need to get all data to run your algo. And `finalize_to_column()` could be called multi times for stream output, so you don't need to store all result in memory.